### ISSUE TITLE
A window-rule that simulates input for the purposes of avoiding idle status

### ISSUE THREAD

**Issue by Kommynct** [hyprwm/Hyprland#7811]
### Description

I have a couple of programs that kill themselves if they've been idle for a while, i've thought about it and realized a window rule in the style of

windowrulev2 = keepalive x input button,class:(whatever)

would be the best solution, where the syntax is

x = ms between keepalives from last time it was focused
input = mouse or simulated keypress, specified by "button" obviously optional for mouse but you could also simulate the back button or something.

**Comment by fufexan** [comment]
Can this not be scripted with ydotool?

**Comment by Kommynct** [comment]
i don't think that would work seamlessly, it would focus the window/bring it to the front constantly, and when in an app that steals focus/layershell, it would break, furthermore this strikes me more as a compositor feature than not.

**Comment by vaxerski** [comment]
Hello there!

This issue has been closed, as we are moving from issues to discussions. More about this can be found [here](https://wiki.hyprland.org/Contributing-and-Debugging/Issue-Guidelines/) and in #9854.

Firstly, please make sure this issue is still relevant in latest Hyprland. If so, we ask you to open a [discussion](https://github.com/hyprwm/Hyprland/discussions) (please read the discussion guidelines first, in the pinned post)

Thank you for your understanding!
